src/Core/Domain/CatalogPriceRule/Command/AddCatalogPriceRuleCommand.php

```diff
@@ -0,0 +1,256 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\Command;
+
+use DateTime;
+use Exception;
+use PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\Exception\CatalogPriceRuleConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Reduction;
+
+/**
+ * Adds new catalog price rule with provided data
+ */
+class AddCatalogPriceRuleCommand
+{
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var int
+     */
+    private $shopId;
+
+    /**
+     * @var int
+     */
+    private $currencyId;
+
+    /**
+     * @var int
+     */
+    private $countryId;
+
+    /**
+     * @var int
+     */
+    private $groupId;
+
+    /**
+     * @var int
+     */
+    private $fromQuantity;
+
+    /**
+     * @var Reduction
+     */
+    private $reduction;
+
+    /**
+     * @var bool
+     */
+    private $includeTax;
+
+    /**
+     * @var float
+     */
+    private $price;
+
+    /**
+     * @var DateTime|null
+     */
+    private $dateTimeFrom;
+
+    /**
+     * @var DateTime|null
+     */
+    private $dateTimeTo;
+
+    /**
+     * @param string $name
+     * @param int $currencyId
+     * @param int $countryId
+     * @param int $groupId
+     * @param int $fromQuantity
+     * @param string $reductionType
+     * @param float $reductionValue
+     * @param int $shopId
+     * @param bool $includeTax
+     * @param float $price
+     */
+    public function __construct(
+        $name,
+        $currencyId,
+        $countryId,
+        $groupId,
+        $fromQuantity,
+        $reductionType,
+        $reductionValue,
+        $shopId,
```

Comment:
Can it only be created for single shop?

Reply:
according to legacy - yes, according to current specs - i don't see shop selection, but waiting for @matks response if it's not missed by mistake.